src/main/drivers/transponder_ir_stm32f30x.c

```diff
@@ -20,121 +20,125 @@
-#define TRANSPONDER_PIN                      GPIO_Pin_8 // TIM16_CH1
-#define TRANSPONDER_PIN_SOURCE               GPIO_PinSource8
-#define TRANSPONDER_TIMER                    TIM16
-#define TRANSPONDER_TIMER_APB2_PERIPHERAL    RCC_APB2Periph_TIM16
-#define TRANSPONDER_DMA_CHANNEL              DMA1_Channel3
-#define TRANSPONDER_IRQ                      DMA1_Channel3_IRQn
-#define TRANSPONDER_DMA_TC_FLAG              DMA1_FLAG_TC3
-#define TRANSPONDER_DMA_HANDLER_IDENTIFER    DMA1Channel3Descriptor
+    #define USE_TRANSPONDER_ON_DMA1_CHANNEL3
+    #define TRANSPONDER_GPIO                     GPIOB
+    #define TRANSPONDER_GPIO_AHB_PERIPHERAL      RCC_AHBPeriph_GPIOB
+    #define TRANSPONDER_GPIO_AF                  GPIO_AF_1
+    #define TRANSPONDER_PIN                      GPIO_Pin_8 // TIM16_CH1
+    #define TRANSPONDER_PIN_SOURCE               GPIO_PinSource8
+    #define TRANSPONDER_TIMER                    TIM16
+    #define TRANSPONDER_TIMER_APB2_PERIPHERAL    RCC_APB2Periph_TIM16
+    #define TRANSPONDER_DMA_CHANNEL              DMA1_Channel3
+    #define TRANSPONDER_IRQ                      DMA1_Channel3_IRQn
+    #define TRANSPONDER_DMA_TC_FLAG              DMA1_FLAG_TC3
+    #define TRANSPONDER_DMA_HANDLER_IDENTIFER    DMA1_CH3_HANDLER
 #endif
 
-void transponderIrHardwareInit(void)
+void transponderIrHardwareInit(transponder_t* transponder_l)
 {
-    TIM_TimeBaseInitTypeDef  TIM_TimeBaseStructure;
-    TIM_OCInitTypeDef  TIM_OCInitStructure;
-    GPIO_InitTypeDef GPIO_InitStructure;
-    DMA_InitTypeDef DMA_InitStructure;
-
-    RCC_AHBPeriphClockCmd(TRANSPONDER_GPIO_AHB_PERIPHERAL, ENABLE);
-
-    GPIO_PinAFConfig(TRANSPONDER_GPIO, TRANSPONDER_PIN_SOURCE,  TRANSPONDER_GPIO_AF);
-
-    /* Configuration alternate function push-pull */
-    GPIO_StructInit(&GPIO_InitStructure);
-    GPIO_InitStructure.GPIO_Pin = TRANSPONDER_PIN;
-    GPIO_InitStructure.GPIO_Mode = GPIO_Mode_AF;
-    GPIO_InitStructure.GPIO_OType = GPIO_OType_PP;
-    GPIO_InitStructure.GPIO_PuPd = GPIO_PuPd_DOWN;
-    GPIO_InitStructure.GPIO_Speed = GPIO_Speed_50MHz;
-    GPIO_Init(TRANSPONDER_GPIO, &GPIO_InitStructure);
-
-    RCC_APB2PeriphClockCmd(TRANSPONDER_TIMER_APB2_PERIPHERAL, ENABLE);
-
-    /* Time base configuration */
-    TIM_TimeBaseStructInit(&TIM_TimeBaseStructure);
-    TIM_TimeBaseStructure.TIM_Period = 156;
-    TIM_TimeBaseStructure.TIM_Prescaler = 0;
-    TIM_TimeBaseStructure.TIM_ClockDivision = 0;
-    TIM_TimeBaseStructure.TIM_CounterMode = TIM_CounterMode_Up;
-    TIM_TimeBaseInit(TRANSPONDER_TIMER, &TIM_TimeBaseStructure);
-
-    /* PWM1 Mode configuration: Channel1 */
-    TIM_OCStructInit(&TIM_OCInitStructure);
-    TIM_OCInitStructure.TIM_OCMode = TIM_OCMode_PWM1;
-    TIM_OCInitStructure.TIM_OutputState = TIM_OutputState_Enable;
-    TIM_OCInitStructure.TIM_Pulse = 0;
-#ifdef TRANSPONDER_INVERTED
-    TIM_OCInitStructure.TIM_OCPolarity = TIM_OCPolarity_Low;
+	TIM_TimeBaseInitTypeDef  TIM_TimeBaseStructure;
```

Comment:
we use 4 spaces, not 8.  please avoid making incorrect indentation changes we can't merge the PR like this and it makes the review process much more difficult.